src/battle/battle_controller_player.c

```diff
@@ -1701,3 +1703,199 @@ void BattleControllerPlayer_SafariBallInput(BattleSystem *bsys, BattleContext *c
+        return 0;
+    }
+
+    if (ov12_02256854(bsys, ctx) == TRUE) {
+        return 0;
+    }
+
+    if (!Battler_CanSelectAction(ctx, ctx->battlerIdAttacker)) {
+        return 0;
+    }
+
+    if (ctx->moveNoCur == MOVE_BIDE && (ctx->battleStatus & BATTLE_STATUS_CHARGE_MOVE_HIT)) {
+        return 0;
+    }
+    
+    if (PlayerProfile_CountBadges(profile) >= 8) {
+        return 0;
+    }
+
+    level = 10;
+    
+    if (PlayerProfile_CountBadges(profile) >= 1) {
+        level = 20;
+    }
+
+    if (PlayerProfile_CountBadges(profile) >= 2) {
+        level = 30;
+    }
+
+    if (PlayerProfile_TestBadgeFlag(profile, 3)) {
+        level = 50;
+    }
+
+    if (PlayerProfile_TestBadgeFlag(profile, 5)) {
+        level = 70;
+    }
+    
+    if (ctx->battleMons[ctx->battlerIdAttacker].level <= level) {
+        return 0;
+    }
+
+    rnd = ((BattleSystem_Random(bsys) & 0xff) * (ctx->battleMons[ctx->battlerIdAttacker].level + level)) >> 8;
+
+    if (rnd < level) {
+        return 0;
+    }
+
+    if (ctx->moveNoCur == MOVE_RAGE) {
+        ctx->battleMons[ctx->battlerIdAttacker].status2 &= ~STATUS2_23;
+    }
+
+    if (ctx->battleMons[ctx->battlerIdAttacker].status & STATUS_SLEEP && (ctx->moveNoCur == MOVE_SNORE || ctx->moveNoCur == MOVE_SLEEP_TALK)) {
+        *script = 254;
+        return 1;
+    }
+    
+    rnd = ((BattleSystem_Random(bsys) & 0xff) * (ctx->battleMons[ctx->battlerIdAttacker].level + level)) >> 8;
+
+    //use a random (useable) move
+    if (rnd < level) {
+        rnd = StruggleCheck(bsys, ctx, ctx->battlerIdAttacker, MaskOfFlagNo(ctx->movePos[ctx->battlerIdAttacker]), -1);
+    
+        if (rnd == 0xF) {
+            *script = 255;
+            return 1;
+        }
+        do {
+            struggleRnd = BattleSystem_Random(bsys) & 3;
+        } while (MaskOfFlagNo(struggleRnd) & rnd);
+    
+        ctx->movePos[ctx->battlerIdAttacker] = struggleRnd;
+        ctx->moveNoTemp = ctx->battleMons[ctx->battlerIdAttacker].moves[ctx->movePos[ctx->battlerIdAttacker]];
+        ctx->moveNoCur = ctx->moveNoTemp;
+        ctx->battlerIdTarget = ov12_022506D4(bsys, ctx, ctx->battlerIdAttacker, ctx->moveNoTemp, 1, 0);
+
+        if (ctx->battlerIdTarget == BATTLER_NONE) {
+            ctx->unk_21A8[ctx->battlerIdAttacker][1] = ov12_02253DA0(bsys, ctx, ctx->battlerIdAttacker);
+        } else {
+            ctx->unk_21A8[ctx->battlerIdAttacker][1] = ctx->battlerIdTarget;
+        }
+
+        *script = 256;
+        ctx->unk_2184 |= 1;
+        
+        return 2; // ???
+    }
+
+    level = ctx->battleMons[ctx->battlerIdAttacker].level - level;
+    rnd = BattleSystem_Random(bsys) & 0xFF;
+
+    //take a nap
+    if (rnd < level && !(ctx->battleMons[ctx->battlerIdAttacker].status & STATUS_ALL) &&
+        GetBattlerAbility(ctx, ctx->battlerIdAttacker) != ABILITY_VITAL_SPIRIT &&
+        GetBattlerAbility(ctx, ctx->battlerIdAttacker) != ABILITY_INSOMNIA &&
+        !(ctx->fieldCondition & (0xF << 8))) {
```

Comment:
is there a constant for this one? field condition constants?

Reply:
I think it's supposed to be uproar? I misread the consts we had documented and thought it was overlapping with the weather ones but I was mistaken looking back.